arangod/Cache/Cache.cpp

```diff
@@ -190,6 +192,24 @@ bool Cache::isBusy() {
   return _metadata.isResizing() || _metadata.isMigrating();
 }
 
+Cache::Inserter::Inserter(Cache& cache, void const* key, std::size_t keySize,
+                          void const* value, std::size_t valueSize,
+                          std::function<bool(Result const&)> retry) {
+  std::unique_ptr<CachedValue> cv{CachedValue::construct(key, keySize, value, valueSize)};
+  if (!cv) {
+    status.reset(TRI_ERROR_OUT_OF_MEMORY);
+  }
+
+  do {
+    status = cache.insert(cv.get());
+    basics::cpu_relax();
```

Comment:
This will call `cpu_relax` unconditionally, even if the insert was successful. This is not ideal and perhaps not intended?

Reply:
Yes, good catch. I'll open a correction.